models/actions/task_list.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/builder"
+)
+
+type TaskList []*ActionTask
+
+func (tasks TaskList) GetJobIDs() []int64 {
+	jobIDsMap := make(map[int64]struct{})
+	for _, t := range tasks {
+		if t.JobID == 0 {
+			continue
+		}
+		jobIDsMap[t.JobID] = struct{}{}
+	}
+	jobIDs := make([]int64, 0, len(jobIDsMap))
+	for jobID := range jobIDsMap {
+		jobIDs = append(jobIDs, jobID)
+	}
+	return jobIDs
```

Comment:
```suggestion
	jobIDs := make([]int64, 0, len(jobIDsMap))
	for _, t := range tasks {
		if t.JobID == 0 {
			continue
		}
		jobIDs = append(jobIDs, t.jobID)
	}
	return jobIDs
```
I don't think wasting that little bit capacity is worth doing everything twice.
Or is `t.JobID == 0` such a common case?
To me, it simply looks like an illegal case we ignore just to be safe…

Reply:
It's for deduplication. And I notice that there's a new way to do that: #21408